File: backend/app/schemas/tasks.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


TaskType = Literal["translation", "spi", "diff"]


class TaskRead(BaseModel):
    task_id: str
    type: TaskType
    status: str
    result_file_id: str | None = None
    error: str | None = None
    created_at: datetime
    updated_at: datetime


class TaskListResponse(BaseModel):
    items: list[TaskRead]

File: frontend/src/types/tasks.ts
export type TaskType = 'translation' | 'spi' | 'diff'

export type UnifiedTask = {
  task_id: string
  type: TaskType
  status: string
  result_file_id: string | null
  error: string | null
  created_at: string
  updated_at: string
}

export type TaskListResponse = {
  items: UnifiedTask[]
}
